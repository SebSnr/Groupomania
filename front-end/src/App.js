import React from "react"
import {BrowserRouter, Switch, Route} from "react-router-dom"
// pages
import HomePage from "./pages/HomePage"
import LoginPage from "./pages/LoginPage"
import ProfilePage from "./pages/ProfilePage"
import MembersPage from "./pages/MembersPage"
import NotFoundPage from "./pages/NotFoundPage"
import ArticlePage from "./pages/ArticlePage"
// utils
import {initialAuth} from "./utils/auth"
import {AuthReducer} from "./utils/auth"

// store authentication data globally
export const AuthContext = React.createContext()

function App() {
	// dispatch authentication data and action
	const [AuthState, dispatchAuthState] = React.useReducer(AuthReducer, initialAuth)

	let routes

	if (AuthState.isAuthenticated) {
		routes = (
			<BrowserRouter>
				<Switch>
					<Route path="/" exact component={HomePage} />
					<Route path="/login" exact component={LoginPage} />
					<Route path="/articles/" exact component={ArticlePage} />
					<Route path="/profile" exact component={ProfilePage} />
					<Route path="/members" exact component={MembersPage} />
					<Route component={NotFoundPage} />
				</Switch>
			</BrowserRouter>
		)
	} else {
		routes = (
			<BrowserRouter>
				<Switch>
					<Route path="/" exact component={HomePage} />
					<Route path="/login" exact component={LoginPage} />
					<Route component={NotFoundPage} />
				</Switch>
			</BrowserRouter>
		)
	}

	return (
		<AuthContext.Provider
			value={{
				AuthState,
				dispatchAuthState,
			}}
		>
			{routes}
		</AuthContext.Provider>
	)
}

export default App
