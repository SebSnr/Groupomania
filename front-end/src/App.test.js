import React from 'react'
import {render, cleanup} from '@testing-library/react'
import 'jest-canvas-mock';
import App from './App'

 afterEach(cleanup)

 it('should take a snapshot', () => {
    const { asFragment } = render(<App />)

    expect(asFragment(<App />)).toMatchSnapshot()
   }) 