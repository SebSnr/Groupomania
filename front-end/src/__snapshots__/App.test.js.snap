// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    height="60"
    options="[object Object]"
    style="background: none; height: 100vh;"
    width="60"
  >
    <svg
      height="500"
      preserveAspectRatio="xMidYMid meet"
      style="width: 100%; height: 100%; transform: translate3d(0,0,0);"
      viewBox="0 0 500 500"
      width="500"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <clippath
          id="__lottie_element_2"
        >
          <rect
            height="500"
            width="500"
            x="0"
            y="0"
          />
        </clippath>
      </defs>
      <g
        clip-path="url(#__lottie_element_2)"
      >
        <g
          style="display: none;"
        >
          <g>
            <path />
          </g>
        </g>
        <g
          opacity="1"
          style="display: block;"
          transform="matrix(0,0,0,0,240.21800231933594,248.42799377441406)"
        >
          <g
            opacity="1"
            transform="matrix(1,0,0,1,171.4149932861328,165.97000122070312)"
          >
            <path
              d=" M5.020999908447266,-165.60000610351562 C4.11299991607666,-165.68099975585938 3.1760001182556152,-165.718994140625 2.2049999237060547,-165.70399475097656 C1.2630000114440918,-165.71800231933594 0.35100001096725464,-165.68499755859375 -0.5320000052452087,-165.60899353027344 C-66.875,-163.48500061035156 -125.16200256347656,-122.08899688720703 -148.21400451660156,-59.507999420166016 C-171.16600036621094,2.802000045776367 -150.23899841308594,76.20500183105469 -98.64199829101562,117.66799926757812 C-46.64400100708008,159.4530029296875 27.746999740600586,165.72000122070312 85.09400177001953,130.54800415039062 C142.343994140625,95.43699645996094 171.1649932861328,28.121000289916992 158.74200439453125,-37.55699920654297 C144.8459930419922,-111.02400207519531 78.33300018310547,-163.1719970703125 5.020999908447266,-165.60000610351562z"
              fill="rgb(255,255,255)"
              fill-opacity="1"
            />
          </g>
        </g>
        <g
          opacity="1"
          style="display: block;"
          transform="matrix(0,0,0,0,242.4239959716797,242.46400451660156)"
        >
          <g
            opacity="1"
            transform="matrix(1,0,0,1,132.5019989013672,132.5019989013672)"
          >
            <path
              d=" M132.2519989013672,0 C132.2519989013672,73.04100036621094 73.04100036621094,132.2519989013672 0,132.2519989013672 C-73.04100036621094,132.2519989013672 -132.2519989013672,73.04100036621094 -132.2519989013672,0 C-132.2519989013672,-73.04100036621094 -73.04100036621094,-132.2519989013672 0,-132.2519989013672 C73.04100036621094,-132.2519989013672 132.2519989013672,-73.04100036621094 132.2519989013672,0z"
              fill="rgb(175,225,240)"
              fill-opacity="1"
            />
          </g>
        </g>
        <g
          opacity="1"
          style="display: block;"
          transform="matrix(0,0,0,0,217.01300048828125,182.48300170898438)"
        >
          <g
            opacity="1"
            transform="matrix(1,0,0,1,154.78500366210938,14.815999984741211)"
          >
            <path
              d=" M31.94700050354004,5.559999942779541 C13.53600025177002,-6.745999813079834 -7.6519999504089355,-13.612000465393066 -29.84600067138672,-14.465999603271484 C-32.439998626708984,-14.5649995803833 -34.571998596191406,-12.427000045776367 -34.571998596191406,-9.831999778747559 C-34.571998596191406,-9.831999778747559 -34.571998596191406,-9.82699966430664 -34.571998596191406,-9.82699966430664 C-34.571998596191406,-7.306000232696533 -32.55400085449219,-5.302999973297119 -30.035999298095703,-5.202000141143799 C-9.678999900817871,-4.388000011444092 9.753999710083008,1.9079999923706055 26.65999984741211,13.168000221252441 C28.756999969482422,14.5649995803833 31.565000534057617,14.128999710083008 33.04600143432617,12.08899974822998 C33.04600143432617,12.08899974822998 33.04800033569336,12.086000442504883 33.04800033569336,12.086000442504883 C34.57099914550781,9.984999656677246 34.104000091552734,7.000999927520752 31.94700050354004,5.559999942779541z"
              fill="rgb(252,249,243)"
              fill-opacity="1"
            />
          </g>
          <g
            opacity="1"
            transform="matrix(1,0,0,1,41.0880012512207,64.16400146484375)"
          >
            <path
              d=" M39.98099899291992,-52.35599899291992 C39.07500076293945,-54.8650016784668 36.2599983215332,-56.051998138427734 33.790000915527344,-55.04399871826172 C-9.531000137329102,-37.34700012207031 -38.80500030517578,4.367000102996826 -40.72800064086914,51.141998291015625 C-40.8380012512207,53.80699920654297 -38.76499938964844,56.051998138427734 -36.097999572753906,56.051998138427734 C-33.577999114990234,56.051998138427734 -31.576000213623047,54.0359992980957 -31.472000122070312,51.51900100708008 C-29.69700050354004,8.355999946594238 -2.684999942779541,-30.13599967956543 37.290000915527344,-46.4640007019043 C39.62300109863281,-47.41699981689453 40.8380012512207,-49.98500061035156 39.981998443603516,-52.354000091552734 C39.981998443603516,-52.354000091552734 39.98099899291992,-52.35599899291992 39.98099899291992,-52.35599899291992z"
              fill="rgb(252,249,243)"
              fill-opacity="1"
            />
          </g>
        </g>
        <g
          style="display: none;"
        >
          <g>
            <path />
          </g>
        </g>
        <g
          style="display: none;"
        >
          <g>
            <path />
          </g>
        </g>
        <g
          style="display: none;"
        >
          <g>
            <path />
          </g>
        </g>
        <g
          style="display: none;"
        >
          <g>
            <path />
          </g>
        </g>
        <g
          style="display: none;"
        >
          <g>
            <path />
          </g>
        </g>
        <g
          style="display: none;"
        >
          <g>
            <path />
          </g>
        </g>
      </g>
    </svg>
  </div>
</DocumentFragment>
`;
